### ISSUE TITLE
Create schema for RESERVED state and REJECTED state CVE

### ISSUE THREAD

**Issue by kurtseifried** [CVEProject/automation-working-group#53]
I suspect it will be easier to create separate schemas for RESERVED and REJECT state CVEs rather than try to create a all encompassing schema with logic for all states. @theall38103 what say you? we'll need to update a bunch of docs and figure out how to break this up. We also need to do this before we can have the CI checking the schema/etc =).

**Comment by theall38103** [comment]
I think it would be good to validate files for all CVE ids, not just those that are public.  That said, I'd like to have it so validation failures are understandable, as opposed to something like : 

`Record did not pass:
{u'description':...} is not valid under any of the given schemas
`

If I have separate schemas for ids in RESERVED and REJECT state and then validate against the appropriate one, I can avoid that. Is there another way?

George

**Comment by theall38103** [comment]
I created two pull requests illustrating the different approaches for the schema. While having a single schema is simpler to manage and run, it makes it much harder to understand what's wrong when validation fails. For example, compare the two sets of output : 

`Record did not pass:
{u'description': {u'description_data': [{u'lang': u'eng', u'value': u'ACTi cameras including the D, B, I, and E series using firmware version A1D-500-V6.11.31-AC fail to properly restrict access to the factory reset page. An unauthenticated, remote attacker can exploit this vulnerability by directly accessing the http://x.x.x.x/setup/setup_maintain_firmware-default.html page. This will allow an attacker to perform a factory reset on the device, leading to a denial of service condition or the ability to make use of default credentials (CVE-2017-3186).'}]}, u'data_type': u'CVE', u'affects': {u'vendor': {u'vendor_data': [{u'product': {u'product_data': [{u'version': {u'version_data': [{u'version_value': u'A1D-500-V6.11.31-AC'}]}, u'product_name': u'ACTi D B I and E series cameras'}]}, u'vendor_name': u'ACTi Corporation'}]}}, u'data_format': u'MITRE', u'problemtype': {u'problemtype_data': [{u'description': [{u'lang': u'eng', u'value': u'CWE-306: Missing Authentication for Critical Function'}]}]}, u'data_version': u'4.0', u'references': {u'reference_data': [{u'url': u'https://www.kb.cert.org/vuls/id/355151\nhttps://cwe.mitre.org/data/definitions/306.html\nhttp://www.securityfocus.com/bid/96720/info\nhttps://twitter.com/hack3rsca/status/839599437907386368\nhttps://twitter.com/Hfuhs/status/839252357221330944'}]}, u'CVE_data_meta': {u'ID': u'CVE-2017-3184', u'ASSIGNER': u'cert@cert.org'}} is not valid under any of the given schemas
`

versus : 

`Record did not pass:
u'https://www.kb.cert.org/vuls/id/355151\nhttps://cwe.mitre.org/data/definitions/306.html\nhttp://www.securityfocus.com/bid/96720/info\nhttps://twitter.com/hack3rsca/status/839599437907386368\nhttps://twitter.com/Hfuhs/status/839252357221330944' does not match u'^(ftp|http)s?://\\S+$'
`

Do people have an opinion on which approach we should take? 

George

**Comment by theall38103** [comment]
Should we close this now that we have separate schemas?

**Comment by mprpic** [comment]
This issue is being closed as obsolete. If you feel it is still relevant to any currently running AWG initiatives, feel free to re-open it or open a new issue. Thank you!
